### ISSUE TITLE
Issue with validation of json request.

### ISSUE THREAD

**Issue by yariks** [laravel/framework#14384]
Hi,

I figured out next issue with validation.

I have next validation rules:
`
[
'answer.value' => 'required'
]
`

Client is posting
```
{
    "answer.value" => "Lorem"
}
```

in request body and with header
`Content-type: application/json`

**What is happen**: validation is pass but data is inappropriate structure.
**What should happen**: validation should return error. Only
```
{
    "answer" => {
        "value": "Lorem"
    }
}
```
structure should be passed.

Can you please fix this?
Thanks

**Comment by themsaid** [comment]
Unfortunately there's no way to fix this in 5.2 without the risk of introducing a breaking change, but in 5.3 `foo.bar` will only grab `foo => [bar]` while `foo\.bar` will grab `foo.bar`.

**Comment by yariks** [comment]
Thanks @themsaid
